src/main/scala/com/snowplowanalytics/indicative/Transformer.scala

```diff
@@ -38,32 +38,52 @@ object Transformer {
    */
   def transform(
     snowplowEvent: String,
-    inventory: Set[InventoryItem]
+    inventory: Set[InventoryItem],
+    unusedEvents: List[String],
+    unusedAtomicFields: List[String],
+    unusedContexts: List[String]
   ): Option[Either[TransformationError, Json]] =
     (for {
       snowplowEvent <- EitherT
         .fromEither[Option](parse(snowplowEvent).leftMap(e => TransformationError(e.message)))
-      indicativeEvent <- EitherT(snowplowJsonToIndicativeEvent(snowplowEvent, inventory))
+      indicativeEvent <- EitherT(
+        snowplowJsonToIndicativeEvent(snowplowEvent, inventory, unusedEvents, unusedAtomicFields, unusedContexts))
     } yield indicativeEvent).value
 
   /**
    * Turns a Snowplow enriched event in a json format into a json ready do be consumed by Indicative.
    * @param snowplowJson Snowplow enriched event in a json format
    * @param inventory a set of inventory items returned by EventTransformer
-   * @return None if the event doesn't contain a user identifying field (user_id,
+   * @return None if the event is in the unusedEvents list, or if it doesn't contain a user identifying field (user_id,
    * client_session_user_id, or domain_userid). Otherwise, it returns either an event in Indicative
    * format or a transformation error.
    */
   def snowplowJsonToIndicativeEvent(snowplowJson: Json,
-                                    inventory: Set[InventoryItem]): Option[Either[TransformationError, Json]] = {
-    val properties = flattenJson(snowplowJson, inventory)
-    val eventName  = extractField(properties, "event_name")
-    val userId = extractField(properties, "user_id")
-      .leftFlatMap(_ => extractField(properties, "client_session_user_id"))
-      .leftFlatMap(_ => extractField(properties, "domain_userid"))
-      .toOption
+                                    inventory: Set[InventoryItem],
+                                    unusedEvents: List[String],
+                                    unusedAtomicFields: List[String],
+                                    unusedContexts: List[String]): Option[Either[TransformationError, Json]] = {
+    val flattenedEvent      = flattenJson(snowplowJson, inventory)
+    val unusedContextFields =
+      // unusedContexts can never be empty because the env var must be set.
+      // But it can contain a single empty string, if no contexts are to be filtered out.
+      if (unusedContexts.head == "") List()
+      else
+        for {
+          context <- unusedContexts
+          keys    <- flattenedEvent.filterKeys(_.startsWith(context)).keySet
+        } yield keys
+    val properties = flattenedEvent -- unusedAtomicFields -- unusedContextFields
+    val eventName  = extractField(flattenedEvent, "event_name")
+    val userId =
+      if (eventName.isRight && unusedEvents.contains(eventName.right.toOption.get)) None // Filter out unused events
```

Comment:
`get` should be avoided as it can result in no such element exception, also I don't see how this line is related to `userId`.

I'd rather we separate concerns and leave user id as:

```scala
    val userId = extractField(properties, "user_id")
      .leftFlatMap(_ => extractField(properties, "client_session_user_id"))
      .leftFlatMap(_ => extractField(properties, "domain_userid"))
      .toOption
```

and add a new variable:

```scala
// a better variable name can surely be found
val eventNameToIgnore = eventName match {
  case Right(n) => Some(()).filterNot(_ => unusedEvents.contains(n))
  case _ => Some(())
}
```

and then

```scala
eventFiltered.flatMap(_ => userId).map { uid =>
```